sw/device/tests/consecutive_irqs/consecutive_irqs_test.c

```diff
@@ -0,0 +1,292 @@
+  // test. TODO - really a DIF routine is needed to reset the uart to defaults.
+  if (!dif_uart_irq_state_clear(uart, kDifUartInterruptRxOverflow)) {
+    uart_send_string("UART failed to clear TX Watermark IRQ!\n");
+    return false;
+  }
+  if (!dif_uart_irq_state_clear(uart, kDifUartInterruptTxEmpty)) {
+    uart_send_string("UART failed to clear TX Empty IRQ!\n");
+    return false;
+  }
+
+  if (!dif_uart_irq_enable(&uart0, kDifUartInterruptRxOverflow,
+                           kDifUartEnable)) {
+    uart_send_string("UART RX overflow IRQ enable failed\n");
+    return false;
+  }
+  if (!dif_uart_irq_enable(&uart0, kDifUartInterruptTxEmpty, kDifUartEnable)) {
+    uart_send_string("UART TX empty IRQ enable failed\n");
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ * Configures all the relevant interrupts in PLIC.
+ */
+static bool plic_configure_irqs(dif_plic_t *plic) {
+  // Set IRQ triggers to be level triggered
+  if (!dif_plic_irq_trigger_type_set(plic, kDifPlicIrqIdUartRxOverflow,
+                                     kDifPlicDisable)) {
+    uart_send_string("PLIC RX overflow trigger type set failed\n");
+    return false;
+  }
+  if (!dif_plic_irq_trigger_type_set(plic, kDifUartInterruptTxEmpty,
+                                     kDifPlicDisable)) {
+    uart_send_string("PLIC TX empty trigger type set failed\n");
+    return false;
+  }
+
+  // Set IRQ priorities to MAX
+  if (!dif_plic_irq_priority_set(plic, kDifPlicIrqIdUartRxOverflow,
+                                 PLIC_PRIORITY_MAX)) {
+    uart_send_string("PLIC priority set for RX overflow failed\n");
+    return false;
+  }
+  if (!dif_plic_irq_priority_set(plic, kDifPlicIrqIdUartTxOverflow,
+                                 PLIC_PRIORITY_MAX)) {
+    uart_send_string("PLIC priority set for TX empty failed\n");
+    return false;
+  }
+
+  // Set Ibex IRQ priority threshold level
+  if (!dif_plic_target_threshold_set(&plic0, PLIC_TARGET, PLIC_PRIORITY_MIN)) {
+    uart_send_string("PLIC threshold set failed\n");
+    return false;
+  }
+
+  // Enable IRQs in PLIC
+  if (!dif_plic_irq_enable_set(plic, kDifPlicIrqIdUartRxOverflow, PLIC_TARGET,
+                               kDifPlicEnable)) {
+    uart_send_string("PLIC Interrupt Enable for RX overflow failed\n");
+    return false;
+  }
+  if (!dif_plic_irq_enable_set(plic, kDifPlicIrqIdUartTxOverflow, PLIC_TARGET,
+                               kDifPlicEnable)) {
+    uart_send_string("PLIC Interrupt Enable for TX empty failed\n");
+    return false;
+  }
+
+  return true;
+}
+
+static bool execute_test(dif_uart_t *uart) {
+  // Force UART RX overflow interrupt, sleep for 0.5s, check if the IRQ has
+  // occured and has been handled appropriately, check that the IRQ has not
+  // been asserted more than once.
+  if (!dif_uart_irq_force(uart, kDifUartInterruptRxOverflow)) {
+    uart_send_string("TEST failed to force RX overflow IRQ\n");
+    return false;
+  }
+  busy_sleep_micros(500000);
```

Comment:
Is spinning actually correct here? It seems to me like you just want to `wait_for_interrupt()` but correct me if I'm wrong.

Reply:
I think waiting is probably fine.

Reply:
Done.